docs/modules/ROOT/pages/explanations/decisions/comp-function-error-handling.adoc

```diff
@@ -0,0 +1,71 @@
+= Composition Function Error Handling
+
+In March 2024 we had a severe outage of VSHN PostgreSQL instances.
+This was due to several unfortunate events including our poor handling of errors in composition functions.
+This document will discuss in detail the problem we face with composition functions when errors are not properly handled.
+
+== Problems
+
+* We don't manage errors in our composition functions (CFs) and we assume Crossplane will take care of everything.
+* Missing managed resources from desired function when an unexpected error happens can have dire consequences, such as removal of the whole service.
+* Returning early from composition function (CF) due to a warning can lead to missing managed resources.
+
+== Requirements
+
+* A comprehensive guide on how we should deal with error management.
+* A solution to missing managed resources from desired function when errors occur.
+
+== Definitions
+
+* Fatal Result error - the error that stops crossplane from reconciliation which means the desired object will not be applied in the cluster.
+Typically safe move which will ensure no changes to the service but we rarely use this type of error.
```

Comment:
Maybe also mention, that usage of the fatal error at the wrong point will lead to deadlocks. E.g. throwing a fatal error, if the observed object is not yet deployed, which will cause the object to never be deployed at all.